hazelcast/src/main/java/com/hazelcast/map/impl/operation/AbstractMultipleEntryBackupOperation.java

```diff
@@ -17,83 +17,75 @@
- * extend this class.
- * <p/>
- * Common functions for these classes can be moved to this class. For now, it only overrides
- * {@link AbstractMultipleEntryOperation#afterRun} method to publish backups of wan replication events.
+ * Provides common backup operation functionality for {@link com.hazelcast.map.EntryProcessor}
+ * that can run on multiple entries.
  */
-abstract class AbstractMultipleEntryBackupOperation extends AbstractMultipleEntryOperation {
+abstract class AbstractMultipleEntryBackupOperation extends MapOperation {
 
-    protected AbstractMultipleEntryBackupOperation() {
-    }
+    protected MapEntries responses;
+    protected EntryBackupProcessor backupProcessor;
+    protected List<WanEventWrapper> wanEventList = Collections.emptyList();
 
-    protected AbstractMultipleEntryBackupOperation(String name, EntryBackupProcessor backupProcessor) {
-        super(name, backupProcessor);
+    public AbstractMultipleEntryBackupOperation() {
     }
 
-    @Override
-    public void afterRun() throws Exception {
-        publishWanReplicationEventBackups();
+    public AbstractMultipleEntryBackupOperation(String name, EntryBackupProcessor backupProcessor) {
+        super(name);
+        this.backupProcessor = backupProcessor;
     }
 
-    protected void publishWanReplicationEventBackups() {
-        for (WanEventWrapper wanEventWrapper : wanEventList) {
-            publishWanReplicationEventBackup(wanEventWrapper.getKey(),
-                    wanEventWrapper.getValue(), wanEventWrapper.getEventType());
-        }
+    protected Predicate getPredicate() {
+        return null;
     }
 
-    protected void publishWanReplicationEventBackup(Data key, Object value, EntryEventType eventType) {
-        if (mapContainer.isWanReplicationEnabled()) {
-            if (REMOVED.equals(eventType)) {
-                mapEventPublisher.publishWanReplicationRemoveBackup(name, key, getNow());
-            } else {
-                final Record record = recordStore.getRecord(key);
-                if (record != null) {
-                    final Data dataValueAsData = toData(value);
-                    final EntryView entryView = createSimpleEntryView(key, dataValueAsData, record);
-                    mapEventPublisher.publishWanReplicationUpdateBackup(name, entryView);
-                }
-            }
-        }
+    protected void setWanEventList(List<WanEventWrapper> wanEventList) {
+        this.wanEventList = wanEventList;
     }
 
     @Override
     protected void writeInternal(ObjectDataOutput out) throws IOException {
         super.writeInternal(out);
-        out.writeInt(wanEventList.size());
-        for (WanEventWrapper wanEventWrapper : wanEventList) {
-            out.writeData(wanEventWrapper.getKey());
-            out.writeData(wanEventWrapper.getValue());
-            out.writeInt(wanEventWrapper.getEventType().getType());
+
+        if (isEmpty(wanEventList)) {
```

Comment:
Can `wanEventList` ever be `null`? The `wanEventList = Collections.emptyList();` looks like it should never be that case, although we have an unprotected setter. If we change the setter to:
```java
protected void setWanEventList(List<WanEventWrapper> wanEventList) {
    if (wanEventList != null) {
        this.wanEventList = wanEventList;
    }
}
```
we could keep the old and simpler `writeInternal()` code. This would also prevent any other accidental NPE on `wanEventList`.